core/src/main/scala/caliban/wrappers/Caching.scala

```diff
@@ -0,0 +1,329 @@
+package caliban.wrappers
```

Comment:
What do we do with `ApolloCaching`? Might be confusing to have both along each other? Also this one doesn't mention Apollo, is it a standard used by others too?

Reply:
Might be wrong, but I think that the main benefit of this one is that since it uses HTTP headers for caching it can be used by any (browser) client. I wonder though whether the `ApolloCaching` would work in cases where this one wouldn't

Reply:
Sorry didn't see the notifications for these comments. Yeah I removed the Apollo prefix because it doesn't actually anything apollo specific, it's using the standard http headers. I'm not sure if it is worth keeping the former I don't think apollo has supported it for a while without customization to the gateway so it may give people the false sense that caching is working (or as we saw recently convince them that caliban is doing some sort of caching).

Reply:
Yeah I'm in favour of deprecating it as well. When I first started using it, I was also under the impression it was caching the responses but found out later that it wasn't 😓

Reply:
Yeah let's deprecate it with a message pointing to the new one 👍